io.sloeber.core/src/io/sloeber/core/api/BoardDescriptor.java

```diff
@@ -918,6 +919,25 @@ public String getUploadCommand(ICConfigurationDescription confdesc) {
 		}
 		String ret = Common.getBuildEnvironmentVariable(confdesc,
 				"A.TOOLS." + upLoadTool.toUpperCase() + "." + action + ".PATTERN", "");
+		
+		// Below we handle the case where the port might have moved. This should be reasonably safe,
+		// since the port value that we have has to have been set, and it has to match a value from
+		// the port list ignoring any numbers on the end. If this specific scenario doesn't occur,
+		// then it will have no effect.
+		String oldComPort = getUploadPort();
+		if (!"".equals(oldComPort)) {
+			String[] comPorts = SerialManager.listComPorts();
+			List<String> ports = Arrays.asList(comPorts);
+			if (!ports.contains(oldComPort)) {
+				for (String port: ports) {
+					if (port.replaceAll("\\d+$", "").equals(oldComPort.replaceAll("\\d+$", ""))) {
```

Comment:
What does this do? Zero or more digits until end of line?

Reply:
Is...  I need to test on Linux, but in the one case I've tested, the serial port seems to be /dev/cu.usbmodem14611, /dev/cu.usbmodem14111, /dev/cu.usbmodem14411, etc.

I've only tested on my Uno, I have a couple of other boards I'll test with too to validate the behaviour.

Reply:
On a mac some USB serial adapters look like this: `/dev/cu.usbserial-A50285BI`.  Not sure if they change when moved to a different usb  port or if the value change depending on what order they are plugged in.....